src/server/game/Handlers/MovementHandler.cpp

```diff
@@ -319,19 +319,24 @@ void WorldSession::HandleMovementOpcodes(WorldPacket & recvData)
 
     recvData.readPackGUID(guid);
 
+    // prevent tampered movement data
+    if (!guid || guid != mover->GetGUID()) {
+        recvData.rfinish();                     // prevent warnings spam
+        return;
+    }
+
     // pussywizard: typical check for incomming movement packets
-    if (!mover || !mover->IsInWorld() || mover->IsDuringRemoveFromWorld() || guid != mover->GetGUID())
+    if (!mover || !(mover->IsInWorld()) || mover->IsDuringRemoveFromWorld() || !(mover->movespline->Finalized()))
```

Comment:
```cpp
if (!mover || !mover->IsInWorld() || mover->IsDuringRemoveFromWorld() || !mover->movespline->Finalized())
```
No need `()`

Reply:
I know, but probably I did it for codacy, because codacy read `!mover` and `!mover->...` like the same condition, probably it was a temporary mistake